ehsm_kms_service/delete_secret_thread.js

```diff
@@ -0,0 +1,131 @@
+const {
+    base64_decode
+} = require('./function')
+
+// Example of setInterval
+var init_interval
+
+// Record timer waiting to run
+var exist_timer = {}
+
+/**
+ * remove a timer from exist timer by key
+ * @param {Object} DB : database control
+ * @param {Object} key : key form exist timer
+ * @returns
+ */
+const removeTimer = (key, DB) => {
+    try {
+        delete exist_timer[key];
+        // An insurance mechanism to ensure that thread init_interval is active
+        init_secret_delete_thread(DB)
+    } catch (error) {
+        console.error('error removeTimer :: ', error)
+    }
+}
+
+/**
+ * Expose the interface externally, scheduled tasks of delete secret
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database control
+ * @param {Object} secretName : The name of the secret. eg. 'secretName01'
+ * @param {Object} plannedDeleteTime : planned delete time for secret eg. '1659931644416'
+ * @returns
+ */
+const startDeleteSecretTask = (DB, appid, secretName, plannedDeleteTime) => {
+    try {
+        let delayMillisecond = plannedDeleteTime - new Date().getTime()
+        let needAdd = true
```

Comment:
what does needAdd mean?  need add what?   Or deleteRequired?

Reply:
it means need to add a deleted task